onionbalance/hs_v3/status.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""
+Provide status over Unix socket
+Default path: /var/run/onionbalance/control
+"""
+import json
+import threading
+import socket
+from socketserver import BaseRequestHandler, ThreadingMixIn, UnixStreamServer
+
+from onionbalance.common import log
+from onionbalance.common.status import BaseStatusSocket
+from onionbalance.hs_v3.onionbalance import Onionbalance
+
+logger = log.get_logger()
+
+
+class StatusSocketHandlerMixin(object):
+    def __init__(self, balance: Onionbalance):
+        self.balance = balance
+
+    def _outputString(self):
+        time_format = "%Y-%m-%d %H:%M:%S"
+        response = {}
+        services_data = []
+        for service in self.balance.services:
+            instances_data = []
+            for instance in service.instances:
+                if not instance.intro_set_modified_timestamp:
+                    instances_data.append({'onionAddress': instance.onion_address + '.onion'})
+                else:
+                    instances_data.append({'onionAddress': instance.onion_address + '.onion',
+                                           'introModified': instance.intro_set_modified_timestamp.strftime(time_format),
+                                           'ipsNum': len(instance.descriptor.intro_set)})
+            last_attempt_first = service.first_descriptor.last_publish_attempt_ts.strftime(time_format) \
```

Comment:
Not a big fan of this multiline `x = y if z else k` construct. Perhaps let's unwrap it into a multiline:
```
if z:
  x = y
 else:
 x = k
```

Reply:
Done.